src/main/java/org/opensrp/web/rest/UploadController.java

```diff
@@ -0,0 +1,378 @@
+			@RequestParam("file") MultipartFile file) {
+		try {
+			List<Map<String, String>> csv_clients = readCSVFile(file);
+			UploadValidationBean validationBean = uploadService.validateFieldValues(csv_clients, eventName, globalID);
+			validationBean.setAnalyzedData(null);
+			return new ResponseEntity<>(gson.toJson(validationBean), HttpStatus.OK);
+		}
+		catch (Exception e) {
+			logger.error("", e);
+		}
+		return new ResponseEntity<>(gson.toJson("An error occurred while reading the file"), HttpStatus.EXPECTATION_FAILED);
+	}
+
+	private void prepareEvent(Event event, String baseEntityID, String providerId, String eventType) {
+		if (event.getFormSubmissionId() == null)
+			event.setFormSubmissionId(UUID.randomUUID().toString());
+
+		if (event.getBaseEntityId() == null)
+			event.setBaseEntityId(baseEntityID);
+
+		event.setDateCreated(new DateTime());
+		event.setEventDate(new DateTime());
+		event.setProviderId(providerId);
+		event.setEventType(eventType);
+		event.setType("Event");
+	}
+
+	private void assignClientUniqueID(Client client, String uniqueIDKey, UniqueIDProvider uniqueIDProvider) {
+		if (StringUtils.isBlank(client.getIdentifier(uniqueIDKey))) {
+			Map<String, String> identifiers = client.getIdentifiers();
+			if (identifiers == null)
+				identifiers = new LinkedHashMap<>();
+
+			identifiers.put(uniqueIDKey, uniqueIDProvider.getNewUniqueID());
+
+			client.setIdentifiers(identifiers);
+		}
+	}
+
+	@RequestMapping(value = "/history", method = RequestMethod.GET, produces = { MediaType.APPLICATION_JSON_VALUE })
+	public ResponseEntity<String> getUploadHistory(HttpServletRequest request) {
+		Integer batchSize = getIntegerFilter(BATCH_SIZE, request);
+		if (batchSize == null)
+			batchSize = 50;
+
+		Integer offset = getIntegerFilter(OFFSET, request);
+		if (offset == null)
+			offset = 0;
+
+		String providerID = getCurrentUser();
+
+		List<UploadBean> uploadBeans = multimediaRepository.getByProviderID(providerID, FILE_CATEGORY, offset, batchSize)
+				.stream()
+				.map(multimedia -> {
+					UploadBean uploadBean = new UploadBean();
+					uploadBean.setFileName(multimedia.getOriginalFileName());
+					uploadBean.setIdentifier(multimedia.getCaseId());
+					uploadBean.setProviderID(multimedia.getProviderId());
+					uploadBean.setUploadDate(multimedia.getDateUploaded());
+					uploadBean.setUrl(multimedia.getCaseId() + "." + FILE_CATEGORY);
+					return uploadBean;
+				})
+				.collect(Collectors.toList());
+
+		return new ResponseEntity<>(
+				gson.toJson(uploadBeans),
+				RestUtils.getJSONUTF8Headers(), HttpStatus.OK);
+	}
+
+	@RequestMapping(value = "/template", method = RequestMethod.GET)
+	public void getUploadTemplate(@RequestParam("event_name") String eventName, HttpServletRequest request,
+			HttpServletResponse response) {
+		String locationID = getStringFilter(LOCATION_ID, request);
+
+		String csvFileName = eventName.replace(" ", "").toLowerCase() + ".csv";
```

Comment:
validate the eventName of csvFileName so that it does not have special characters, check Multimedia resource

Reply:
done